src/io/src/DataLoaderBridge.cpp

```diff
@@ -0,0 +1,289 @@
+#include <rs/io/DataLoaderBridge.h>
+
+//PCL include
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/io/ply_io.h>
+
+#include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/path.hpp>
+#include <boost/progress.hpp>
+#include <boost/optional/optional.hpp>
+
+#include <functional>
+#include <iterator>
+#include <sstream>
+
+
+using namespace uima;
+
+namespace fs = boost::filesystem;
+
+DataLoaderBridge::DataLoaderBridge(const boost::property_tree::ptree& pt) : CamInterface(pt){
+  isLoop = true;
+
+  isCloudFile = true;
+  isRGBFile = true;
+  isDepthFile = true;
+
+  haveCloud = false;
+  haveRGB = false;
+  haveDepth = false;
+
+  iterator = 0;
+
+
+  if (this->readConfig(pt))
+    this->_newData = true;
+  else
+    this->_newData = false;
+
+  if (this->frameRate > 0 && this->_newData) {
+    auto worker = std::bind(&DataLoaderBridge::updateTimerWorker, this,
+      std::chrono::milliseconds(std::lround(1000 / this->frameRate)));
+    this->updateTimerThread = std::thread(worker);
+  }
+}
+
+DataLoaderBridge::~DataLoaderBridge() {}
```

Comment:
Could yo please add the following code to destructor:
```
this->done = true;
if (this->updateTimerThread.joinable())
  this->updateTimerThread.join();
```
To abort timer thread correctly